### ISSUE TITLE
macos: receiving empty UDP datagram fails on 10.15 (catalina)

### ISSUE THREAD

**Issue by santigimeno** [libuv/libuv#2526]
It seems an OS X issue but I post it in case someone has an idea.

Sample code that reproduces the issue: 
```c
#include "uv.h"
#include <assert.h>

static uv_udp_t server;
static uv_udp_t client;

static void alloc_cb(uv_handle_t* handle,
                     size_t suggested_size,
                     uv_buf_t* buf) {
  static char slab[65536];
  buf->base = slab;
  buf->len = sizeof(slab);
}

static void recv_cb(uv_udp_t* handle,
                       ssize_t nread,
                       const uv_buf_t* rcvbuf,
                       const struct sockaddr* addr,
                       unsigned flags) {
  int r;

  assert(nread == 0);
  r = uv_udp_recv_stop(handle);
  uv_close((uv_handle_t*)handle, NULL);
}

static void send_cb(uv_udp_send_t* req, int status) {
  int r;

  assert(req != NULL);
  assert(status == 0);
  uv_close((uv_handle_t*)req->handle, NULL);
}

int main() {
  struct sockaddr_in addr;
  uv_udp_send_t req;
  uv_buf_t buf;

  uv_ip4_addr("0.0.0.0", 9876, &addr);
  uv_udp_init(uv_default_loop(), &server);
  uv_udp_bind(&server, (const struct sockaddr*) &addr, 0);
  uv_udp_recv_start(&server, alloc_cb, recv_cb);
  uv_ip4_addr("127.0.0.1", 9876, &addr);

  uv_udp_init(uv_default_loop(), &client);

  buf = uv_buf_init("", 0);

  uv_udp_send(&req, &client, &buf, 1, (const struct sockaddr*) &addr, send_cb);

  uv_run(uv_default_loop(), UV_RUN_DEFAULT);

  return 0;
}
```

Refs: https://github.com/nodejs/node/issues/30030

**Comment by vtjnash** [comment]
Also ran into this in JuliaLang (https://github.com/JuliaLang/julia/issues/33582)

**Comment by santigimeno** [comment]
It looks like `kqueue` does not notify about the message received :man_shrugging:

**Comment by vtjnash** [comment]
Still appears to be a kernel regression with @apple's latest release (was fixed in El Capitan and broken again in Catalina). No response yet to the Radars filed against this.

**Comment by Keno** [comment]
This was fixed in macOS 10.15.4.
